cmd/utils/flags.go

```diff
@@ -929,11 +929,27 @@ func setEthash(ctx *cli.Context, cfg *eth.Config) {
 	}
 }
 
-func checkExclusive(ctx *cli.Context, flags ...cli.Flag) {
+func checkExclusive(ctx *cli.Context, args ...interface{}) {
 	set := make([]string, 0, 1)
-	for _, flag := range flags {
-		if ctx.GlobalIsSet(flag.GetName()) {
-			set = append(set, "--"+flag.GetName())
+	for i, arg := range args {
+		if flag, ok := arg.(cli.Flag); ok {
+			if ctx.GlobalIsSet(flag.GetName()) {
+				// Check if next arg extends current
+				if len(args) > i+1 {
+					switch option := args[i+1].(type) {
+					case cli.Flag:
+						set = append(set, `"--`+flag.GetName()+`"`)
+					case string:
+						if ctx.GlobalString(flag.GetName()) == option {
+							set = append(set, `"--`+flag.GetName()+` `+option+`"`)
+						}
+					default:
+						Fatalf("Received wrong type in checkExclusive")
```

Comment:
I think a panic here would be better. It shouts louder that something was coded wrong:

```go
panic("unsupported flag argument type")
```